tests/Unit/SDK/Common/Dev/Compatibility/UtilTest.php

```diff
@@ -12,24 +13,86 @@
  */
 class UtilTest extends TestCase
 {
-    public function test_trigger_class_deprecation_notice(): void
+    public function tearDown(): void
     {
-        $this->expectNotice();
+        Util::setErrorLevel();
+    }
+
+    /**
+     * @dataProvider errorLevelProvider
+     */
+    public function test_set_error_level(int $level): void
+    {
+        Util::setErrorLevel($level);
+
+        $this->assertSame(
+            $level,
+            Util::getErrorLevel()
+        );
+    }
+
+    /**
+     * @dataProvider errorLevelProvider
+     */
+    public function test_set_error_level_throws_exception_on_in_correct_level(int $level): void
```

Comment:
this doesn't use the data provider

Reply:
oops, you are right. That was a late and sloppy addition :)